### ISSUE TITLE
Extend auto user provisioning to other databases

### ISSUE THREAD

**Issue by r0mant** [gravitational/teleport#27323]
In https://github.com/gravitational/teleport/pull/25614 we implemented automatic user provisioning for self-hosted PostgreSQL and RDS PostgreSQL.

Let's extend this feature to support other database types and protocols. In the order of priority:

```[tasklist]
- [ ] Add auto user provisioning for RDS MySQL.
- [ ] Add auto user provisioning for self-hosted MySQL.
- [ ] Add auto user provisioning for GCP PostgreSQL and MySQL.
- [ ] Add auto user provisioning for Azure PostgreSQL and MySQL.
- [ ] Add auto user provisioning for self-hosted MongoDB and Atlas.
- [ ] ...
```

**Comment by philip-teleport** [comment]
A prospect has a requirement for auto user provisioning for Oracle. Can this also be considered?

**Comment by greedy52** [comment]
> A prospect has a requirement for auto user provisioning for Oracle. Can this also be considered?

yes, we are planning to work on auto-user provisioning for Oracle (later this Q or next Q) after Oracle RDS support. 

cc @Tener

**Comment by phall-teleport** [comment]
A customer is asking for this for GCP hosted Cloud SQL (PGSQL) as well is this being considered?

**Comment by greedy52** [comment]
> A customer is asking for this for GCP hosted Cloud SQL (PGSQL) as well is this being considered?

no plan at the moment. Also IAM service accounts are used to access Cloud SQL so will be tricky to implement auto-user provisioning.

**Comment by AlgirdasEidukaitis** [comment]
We would love to see this added for GCP MySQL and PostgreSQL ❤️
